pyntcloud/core_class.py

```diff
@@ -625,9 +626,76 @@ def _update_points(self, df):
         self.xyz = self.__points[["x", "y", "z"]].values
         self.centroid = self.xyz.mean(0)
 
+
+    def to_pyvista(self, mesh=False, use_as_color=("red", "green", "blue")):
+        """Convert this PyntCloud to a PyVista mesh object"""
+        try:
+            import pyvista as pv
+        except ImportError:
+            raise ImportError('PyVista must be installed. Try `pip install pyvista`')
+        if mesh and self.mesh is not None:
+            mesh = self.mesh[["v1", "v2", "v3"]].values
+        else:
+            mesh = None
+        # Either make point cloud or triangulated mesh
+        if mesh is not None:
+            # Update cells of PolyData
+            types = np.full(len(mesh), 3, dtype=int)
+            faces = np.insert(mesh, 0, types, axis=1)
+            poly = pv.PolyData(self.xyz, faces)
+        else:
+            poly = pv.PolyData(self.xyz)
+
+        avoid = ["x", "y", "z"]
+        # add scalar arrays
+        if all(c in self.points.columns for c in use_as_color):
+            colors = self.points[list(use_as_color)].values
+            poly.point_arrays["RGB"] = colors
+            avoid += list(use_as_color)
+        # Add other arrays
+        for name in self.points.columns:
+            if name not in avoid:
+                poly.point_arrays[name] = self.points[name]
+
+        return poly
+
+
+    @classmethod
+    def from_pyvista(cls, poly_data):
+        """Load a PyntCloud mesh from a PyVista mesh"""
+        try:
+            import pyvista as pv
+        except ImportError:
+            raise ImportError('PyVista must be installed. Try `pip install pyvista`')
+        if not isinstance(poly_data, pv.PolyData):
+            raise TypeError('Type {} not yet supported for conversion.'.format(type(poly_data)))
+        # Grab the point coordinates
+        points = pd.DataFrame(data=poly_data.points, columns=["x", "y", "z"])
+        # Get the triangulated mesh if present
+        mesh = None
+        if poly_data.faces.ndim > 1:
+            mesh = poly_data.faces
+            if not np.all(3 == mesh[:,0]):
+                raise AssertionError('This mesh is not triangulated. Try triangulating the mesh before passing to PyntCloud.')
+            mesh = pd.DataFrame(data=mesh[:, 1:], columns=['v1', 'v2', 'v3'])
+        # pass scalar arrays to PyntCloud mesh
+        scalars = poly_data.point_arrays
+        for name, array in scalars.items():
+            if name in "RGB":
+                points["red"] = array[:,0]
+                points["green"] = array[:,1]
+                points["blue"] = array[:,2]
+            elif array.ndim < 2:
+                points[name] = array
+            else:
+                # Multicomponent arrays aren't supported by PyntCloud?
```

Comment:
It is a Pandas DataFrame limitation: 

https://pandas.pydata.org/pandas-docs/version/0.18/generated/pandas.DataFrame.html

Reply:
Hm, perhaps we should throw a warning here? I work with multicomponent arrays a lot and I could see users getting confused when those arrays aren't copied over to the `PyntCloud` object. 

Or maybe we could use the array name and add `*-0`, `*-1`, ...  `*-n` for each component?

Reply:
Could you share a minimal example of a multicomponent array ?

Reply:
Here's a simple case using surface normals which are generated with this geometric object in PyVista:

```py
>>> import pyvista as pv
>>> poly = pv.Sphere()
>>> multi_comp_array = poly['Normals']
>>> multi_comp_array
array([[ 0.        ,  0.        ,  1.        ],
       [ 0.        ,  0.        , -1.        ],
       [ 0.10811902,  0.        ,  0.99413794],
       ...,
       [ 0.31232402, -0.06638652, -0.9476532 ],
       [ 0.21027282, -0.04469487, -0.97662055],
       [ 0.10575636, -0.02247921, -0.99413794]], dtype=float32)
>>> multi_comp_array.shape
(842, 3)
>>> poly.n_points
842
```

But perhaps the multicomponent array could have 2 or more componenents (not just 3) and this components could represent anything (not just surface normals, but perhaps a vector field for stress on the surface of the mesh or some physical property)

I think the easiest path forward would be to take the name of the array (`'Normals'` in this case) and add on `-0`, `-1`, and `-2` for each component when adding it to the `PyntCloud DataFrame.

Reply:
Should be supported now. I added a special case like the RGB you added for "Normals" (expecting that pyvista usually name normals like that and beacause PyntCloud expects normals to be named in a specific way)

Reply:
https://github.com/daavoo/pyntcloud/blob/master/pyntcloud/core_class.py#L691

Reply:
Maybe the `to_pyvista` method should also be adapted to handle normals this way